packages/http/src/Adaptor.js

```diff
@@ -60,13 +60,14 @@ export function execute(...operations) {
  * @function
  * @param {string} method - The HTTP method to use.
  * @param {string} path - Path to resource. Can be an absolute URL if baseURL is NOT set on `state.configuration`.
- * @param {RequestOptions} params - Query, Headers and Authentication parameters
+ * @param {object} data - Body data to append to the request. JSON will be converted to a string.
+ * @param {RequestOptions} options - Query, Headers and Authentication parameters
  * @param {function} callback - (Optional) Callback function
  * @state {HttpState}
  * @returns {Operation}
  */
-export function request(method, path, params, callback) {
-  return sendRequest(method, path, params, callback);
+export function request(method, path, data, options, callback) {
```

Comment:
hmm, a bit surprised by this change :thinking: Can you say more about why this function changed?

Is it because of the `RequestOptions` typedef? If so, can you think of any alternative solutions?

Reply:
I changed it to match post, but I can let it look the way it was before with only changing `params` to `options`

Reply:
Yes please

This is a general function and could be used for anything

`get` and `post` are specialists - they're designed to do one thing very well.

The signatures must reflect this :)